R/classicProcess.R

```diff
@@ -1926,6 +1928,54 @@ ClassicProcess <- function(jaspResults, dataset = NULL, options) {
   }
 }
 
+
+.procRsquared <- function(options, modelsContainer) {
+  # adapted from .semRsquared() from jaspSem/R/sem.R (one table for several models)
+  if (!options[["rSquared"]] || !is.null(modelsContainer[["rsquared"]])) return()
+  
+  # retrieve model objects
+  procResults <- lapply(options[["processModels"]], function(mod) modelsContainer[[mod[["name"]]]][["fittedModel"]]$object)
+  modelNames <- sapply(options[["processModels"]], function(mod) mod[["name"]])
+  
```

Comment:
```suggestion
  resultIsValid <- sapply(procResults, function(mod) inherits(mod, "lavaan") && mod@Options[["do.fit"]])
  procResults <- procResults[resultIsValid]
  modelNames <- modelNames[resultIsValid]
```

If the estimation for a model fails or a model is not complete, yet, the model should be excluded from the table I believe. See also `.procModelSummaryTable`.